File: python/rename_files.py
# -*- coding: utf-8 -*-

# Rename multiple files
# Putting a sequence of numbers at the beginning of the file name

import os

# Path where are the files that it will renames
path = "./my-files"

if os.path.isdir(path):
	# Loop over a list and get the index (using enumerate) and the value (name of file)
	for num, name in enumerate(os.listdir(path), start=1):
		new_name = str(num) + " - " + str(name)

		os.rename(path+"/"+name, path+"/"+new_name)
		print("File '" + name + "' changed to '" + new_name + "'")

File: php/rename_files.php
<?php
/* 
* Rename multiple files
* Putting a sequence of numbers at the beginning of the file name
*/


	// Function to rename all the files in a directory
	function renameFiles($path) {

		if (!is_dir($path)) {
			echo "'" . $path . "' is not a directory!" . "\r\n";
			exit;
		}
		
		$files = scandir($path);
		
		$num = 1;

		echo 'Renaming files...' . "\r\n \r\n";

		foreach ($files as $file) {
			

			if ($file != '.' && $file != '..') {
				$newname = $num . ' - ' . $file;
				
				echo $newname . "\r\n";

				if ( $file != $newname ) {
					rename($path . $file, $path . $newname);  
				}
	
				$num++;
			}
		}
		
		echo "\r\n" . 'Done!' . "\r\n";
	}


	// Path where are the files that it will renames
	$path = "./my-files/";
	renameFiles($path);




